### DISCUSSION TITLE
Blog starter template should use Astro's image system for hero images and use them in og:image

### DISCUSSION THREAD

**Discussion by tmcw** [withastro/roadmap#791] [Proposal]
### Body

# Summary

Right now, the blog starter for Astro has an idea  of a `heroImage` at the top of each blog post. But these images in are in the `/public/` dir, not in `src/content`, and they don't get resized and optimized by Astro. It'd be much better if they were content, so that Astro could make sure that they were efficient.

# Background & Motivation

This was a surprise, and seems like a straightforward improvement for the starter template

# Goals

- Make images served by the starter template smaller and more optimized
- Have the astro starters demonstrate best practices

**Comment by NicholasDewberryOfficial** [comment]
Just noticed this too. 

Most of the detail containing the stuff for the blogpost is in BlogPost.astro - I don't see anything that would us from moving it to /src/ 

I could make a fork, just to show off how the starter template should look.

**Comment by florian-lefebvre** [comment]
Feel free to submit a PR! I also think we should take advantage of the image features
